MobileConnectionLibrary/MobileConnectionLibrary/TelephoneNumberComparer.cs

```diff
@@ -0,0 +1,22 @@
+﻿using MobileConnectionLibrary;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace MobileConnection
+{
+
+    public class TelephoneNumberComparer : IComparer<Subscriber>
```

Comment:
Имя класса неудачное.  Логика такая: кого сравниваем? Абонентов. Как сравниваем? По номеру телефона.
```suggestion
    public class SubscriberComparerByTelephoneNumber : IComparer<Subscriber>
```